content/api/commands/mount.md

```diff
@@ -0,0 +1,226 @@
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Mounts a React node
+       * @param jsx React Node to mount
+       * @param options Additional options to pass into mount
+       */
+      mount(
+        jsx: React.ReactNode,
+        options?: MountOptions
+      ): Cypress.Chainable<MountReturn>
+    }
+  }
+}
+```
+
+```ts
+import { mount } from '@cypress/vue'
+type MountParams = Parameters<typeof mount>
+
+declare global {
+  namespace Cypress {
+    type ComponentParam = MountParams[0] | JSX.Element
+    type OptionsParam = MountParams[1]
+    interface Chainable {
+      /**
+       * Helper mount function for Vue Components
+       * @param component Vue Component or JSX Element to mount
+       * @param options Options passed to Vue Test Utils
+       */
+      mount(component: ComponentParam, options?: OptionsParam): Chainable<any>
+    }
+  }
+}
+```
+
+:::
+
+The typings will need to be in a location that any code can access in the
+application. We recommend adding them to a `cypress.d.ts` file at the root of
+the application and updating any tsconfig.json files to reference the typings
+file in the `include` option:
+
+```json
+"include": ["./src", "cypress.d.ts"]
+```
+
+## Additional Mount Commands
+
+You can create additional mount commands to fit your needs. For instance, there
+might be times when you need to do additional setup for certain tests that other
+tests don't need. In this case, it is good to create a new command for those
+tests.
+
+Below are some examples for specific use cases.
+
+## React Examples
+
+### React Router
+
+If you have a component that consumes a hook or component from React Router, you
+will need to make sure the component has access to a React Router provider.
+Below is a sample mount command that uses `MemoryRouter` to wrap the component.
+Setup props for `MemoryRouter` can be passed in the options param as well:
+
+```jsx
+import { mount } from '@cypress/react'
+import { MemoryRouter } from 'react-router-dom'
+
+Cypress.Commands.add('mountWithRouter', (component, options) => {
+  const { routerProps = { initialEntries: ['/'] }, ...mountOptions } = options
+
+  const wrapped = <MemoryRouter {...routerProps}>{component}</MemoryRouter>
+
+  return mount(wrapped, mountOptions)
+})
+```
+
+You can pass in custom props for the `MemoryRouter` in the `options` param.
+
+Example usage:
+
+```jsx
+it('Logout link should appear when logged in', () => {
+  cy.mountWithRouter(<Navigation loggedIn={true} />, {
+    routerProps: {
+      initialEntries: ['/home'],
+    },
+  })
+  cy.contains('Logout').should('exist')
+})
+```
+
+### Redux
+
+To use a component that consumes state or actions from a Redux store, you can
+create a `mountWithRedux` command that will wrap your component in a Redux
+Provider:
+
+```jsx
+import { mount } from '@cypress/react'
+import { Provider } from 'react-redux'
+import { store } from '../../redux/store'
```

Comment:
I'm not sure we'd ever want to recommend importing a singleton store in a test, because there's no way to reset the store back to a default value before each test. I would recommend exporting some kind of factory for the store so that it can be called in the custom command or beginning of each test.

See:

- https://github.com/cypress-io/cypress-component-testing-examples/commit/e77563b77ba793b572b813baf469f197dedcbce8
- https://github.com/cypress-io/cypress-component-testing-examples/commit/547a22245a447a32290ed05df805ccbfc047b9e2#diff-7afb312e0f41ed345359e3804985d887e63b1446dcd2f095ac3992440bd63e2e

Reply:
good catch